Source/WebCore/css/query/ContainerQueryFeatures.cpp

```diff
@@ -42,6 +42,15 @@ namespace WebCore::CQ {
 
 using namespace MQ;
 
+enum class UsesSVGZoomRulesForLength : bool { No, Yes };
+
+static LayoutUnit unscaledSizeForPrincipleBox(const Style::PreferredSize& computedSize, LayoutUnit usedSize, UsesSVGZoomRulesForLength usesSVGZoomRulesForLength, const Function<float()>& zoomFunction)
+{
+    if (usesSVGZoomRulesForLength == UsesSVGZoomRulesForLength::Yes || computedSize.isAuto() || computedSize.isPercent())
+        return usedSize;
+    return LayoutUnit { usedSize / zoomFunction() };
+}
```

Comment:
i think you can send the whole style object here instead?

```
static LayoutUnit unscaledSizeForPrincipleBox(const Style::PreferredSize& computedSize, LayoutUnit usedSize, const RenderStyle& style) {
    if (style.useSVGZoomRulesForLength() || ...)
        return usedSize;

    return LayoutUnit { usedSize / style.usedZoom() };
}
```

and then at call-site

`auto width = unscaledSizeForPrincipleBox(style.width(), renderer.contentBoxWidth(), style);`

Reply:
let's make the name of this routine more obvious perhaps `computedSizeWithoutZoom` - because that's what it actually returns.

Reply:
I don't think that's what it is returning